### DISCUSSION TITLE
SSH keys support

### DISCUSSION THREAD

**Discussion by KaKi87** [catpuppyapp/PuppyGit#2] [Ideas]
Hi,

It would be nice to be able to authenticate by SSH keys.

Thanks

**Comment by Bandeapart1964** [comment]
sure, already supported, its under testing

**Reply by KaKi87** [reply]
Alright, I'm looking forward to see it in a release. Thanks !

**Reply by Bandeapart1964** [reply]
> Alright, I'm looking forward to see it in a release. Thanks !

hi, try the latest release, supported ssh now😉

**Reply by KaKi87** [reply]
I don't see any update on IzzyOnDroid yet, I'll wait. Thanks

**Reply by KaKi87** [reply]
Hi,
I see the *Username* input was renamed to *Username/Private Key* in *Credentials Manager*, but how to generate a key pair ?
Thanks

**Reply by Bandeapart1964** [reply]
> Hi,
> I see the *Username* input was renamed to *Username/Private Key* in *Credentials Manager*, but how to generate a key pair ?
> Thanks

https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#generating-a-new-ssh-key

**Reply by KaKi87** [reply]
Well, on desktop, yes of course, but not on Android 😅

**Reply by Bandeapart1964** [reply]
on android you can try Termux gen the keys, see: https://github.com/rishabhrpg/sshd-in-termux?tab=readme-ov-file#on-your-androidos-device

**Reply by KaKi87** [reply]
Well, I could also use git on Termux, but the whole point of using an app is not requiring a terminal on mobile

**Reply by Bandeapart1964** [reply]
so, what are you expect? generate a ssh key in a git client? I don't think is a good idea.

**Reply by Bandeapart1964** [reply]
I created a issue, if ppl expect generate ssh keys on puppygit, i'll implement
https://github.com/catpuppyapp/PuppyGit/issues/6

**Reply by Bandeapart1964** [reply]
check the issue #6 , I create a app for manage ssh keys for instead generate keys in puppygit

**Reply by KaKi87** [reply]
Why a separate app ?
